cEP-0018.md

```diff
@@ -22,6 +22,38 @@ they can write advanced linting bears. This will aim at supporting the more
 flexible visitor based method of AST traversal (as opposed to listener based
 mechanisms).
 
+ANTLR has a `HiddenTokenStream` which recieves all the tokens irrelevant to the
+grammar such as whitespaces. These are not passed on to the parser and as a
+result are not returned when simply fetching text from a node in parse tree.
+
+Consider the code:
+
+```python
+var_x = 5
+if var_x:
+    print('Variable x is', var_x)
+else:
+    print('Variable x is not set')
+```
+
+After converting this to parsetree and fetching source back using
+[this](https://gist.github.com/virresh/643cd1a2d9149668980814d970f98fbc#file-roundtripper-py-L13)
+code we get the following output:
+
+```python
+var_x=5
+ifvar_x:
```

Comment:
this is misleading.  `if` and `var_x` are two nodes in the parse tree?  So you can choose to always add whitespace between nodes to visualise the parse tree has distinct nodes.

Reply:
Yes, this is always a possible thing, 
> The alternative we have is we add whitespaces on our  own according to the column number and line number (but this obviously is wrong, since we simply won't know what the actual whitespace used was for, was it a tab or a space)

But since we are adding a white-space ourselves, this is not feasible in all scenarios, for e.g when adding indentation in the line below the `if` statement will not differentiate between spaces or tabs, this information is lost permanently :sweat_smile: